common/services/files/allowed_document_extensions.py

```diff
@@ -0,0 +1,30 @@
+from flask import current_app
+from invenio_records_resources.services.files.components.base import (
+    FileServiceComponent,
+)
+
+from invenio_records_resources.services.errors import FailedFileUploadException
+from flask_babel import _
+
+
+class AllowedDocumentExtensionsComponent(FileServiceComponent):
+    def init_files(self, identity, id, record, data):
+        allowed_extensions = current_app.config["ALLOWED_DOCUMENT_FILE_EXTENSIONS"]
+
+        for file_data in data:
+            file_name = file_data["key"]
+            if not file_name.endswith(tuple(allowed_extensions)):
+                raise InvalidFileExtensionException(file_name)
+
+
+class InvalidFileExtensionException(Exception):
+    """File has invalid extension. Refer to ALLOWED_DOCUMENT_FILE_EXTENSIONS in invenio.cfg for allowed extensions."""
+
+    def __init__(self, file_key):
+        """Constructor."""
+        super().__init__(
+            _("Failed to upload '{file_key}'. Invalid extension.").format(
```

Comment:
Please add a list of allowed extensions to the exception's message, so that user can see in the UI which extensions are allowed.